core/src/main/java/org/infinispan/context/impl/FlagBitSets.java

```diff
@@ -65,4 +65,18 @@ public static Flag extractStateTransferFlag(InvocationContext ctx, FlagAffectedC
       }
       return null;
    }
+
+   /**
+    * Retrieves the values for the given flags and then sets those flags on the command. This can be useful
+    * if subsequent users of this command know that an upstream operation has already handled a given flag
+    * @param command the command to get the flags from and then set
+    * @param flagsToSet the flags to get and set
+    * @return the flags that were set
+    */
+   public static long getAndSetFlags(FlagAffectedCommand command, long flagsToSet) {
+      long originalFlags = command.getFlagsBitSet();
+      long setFlags = originalFlags & flagsToSet;
+      command.setFlagsBitSet(originalFlags | flagsToSet);
```

Comment:
I would expect a method named `getAndSet` to set all the flags and to return all the previous flags, without any change. A more appropriate name would be `getFlagsIntersectionAndSetFlagsUnion` ;)

I'd also prefer having it as a default method in `FlagAffectedCommand`, and using `EnumUtil` methods instead of bit operations.